dragonfly/engines/base/engine.py

```diff
@@ -129,6 +129,10 @@ def _build_grammar_wrapper(self, grammar):
     #-----------------------------------------------------------------------
     # Methods for working with grammars.
 
+    def window_change(self, executable, title, handle):
```

Comment:
We don't really need another engine method for this. The `grammars` property can be used instead:

```Python
for grammar in get_engine().grammars:
    grammar.process_begin(executable, title, handle)

```